firebase_admin/_auth_utils.py

```diff
@@ -275,6 +277,135 @@ def validate_provider_ids(provider_ids, required=False):
         validate_provider_id(provider_id, True)
     return provider_ids
 
+
+def validate_mfa_config(mfa_config):
+    """Validates the specified multi factor configuration."""
+
+    mfa_config_payload = {}
+    # validate multiFactorConfig
+    if not isinstance(mfa_config, dict):
+        raise ValueError(
+            'multiFactorConfig should be of valid type MultiFactorConfig')
+
+    #validate MFA Config keys
+    valid_mfa_config_keys = set(['state', 'factorIds', 'providerConfigs'])
+    mfa_config_keys = set(mfa_config.keys())
+    for key in mfa_config_keys:
+        if key not in valid_mfa_config_keys:
+            raise ValueError('{0} is not a valid MultiFactorConfig parameter'.format(key))
+
+    # validate state in MFA config
+    if 'state' not in mfa_config:
+        raise ValueError('multiFactorConfig.state should be defined')
+    state = mfa_config['state']
+    if not isinstance(state, str) or state not in VALID_STATES:
+        raise ValueError(
+            'multiFactorConfig.state must be either "ENABLED" or "DISABLED"')
+    mfa_config_payload['state'] = state
+
+    # validate factorIds if defined
+    if 'factorIds' in mfa_config:
+        factor_ids = mfa_config['factorIds']
+        if not isinstance(factor_ids, list) or not factor_ids:
+            raise ValueError(
+                'multiFactorConfig.factorIds must be a defined list of AuthFactor type strings')
+        for factor_id in factor_ids:
+            if not isinstance(factor_id, str) or factor_id not in VALID_AUTH_FACTOR_TYPES:
+                raise ValueError(
+                    'factorId must be in ' + str(VALID_AUTH_FACTOR_TYPES))
+        mfa_config_payload['enabledProviders'] = factor_ids
+
+    # validate providerConfigs if defined
+    if 'providerConfigs' in mfa_config:
+        provider_configs = mfa_config['providerConfigs']
+        provider_configs_payload = []
+        if not isinstance(provider_configs, list) or not provider_configs:
+            raise ValueError(
+                'multiFactorConfig.providerConfigs must be a valid list of providerConfig types')
+        for provider_config in provider_configs:
+            provider_config_payload = {}
+            if not isinstance(provider_config, dict) or not provider_config:
+                raise ValueError(
+                    'multiFactorConfig.providerConfigs must be a valid list of '
+                    'providerConfig types')
+
+            #validate Provider Config keys
+            valid_provider_config_keys = set(['state', 'totpProviderConfig'])
+            provider_config_keys = set(provider_config.keys())
+            for key in provider_config_keys:
+                if key not in valid_provider_config_keys:
+                    raise ValueError('{0} is not a valid ProviderConfig parameter'.format(key))
```

Comment:
This block of code seems repeat itself multiple time.  Consider modify this to a generic method and put in auth_utils.

E.g.
in _auth_utils.py:
```
def validate_attribute(valid_attribute_list, input, propertyName)
  input_keys = set(input.keys())
  for key in input_keys:
     if key not in valid_attribute_list:
         raise ValueError(...)
```

and everytime we can just call with one line code `validate_attribute(valid_provider_config_keys, provider_config_keys, 'ProviderConfig')`;

Reply:
Ack, this definitely looks more efficient. Thanks for suggesting this!